gap/SkeletalFinSetsForCAP.gd

```diff
@@ -49,6 +49,8 @@ DeclareAttribute( "AsList",
         IsSkeletalFiniteSet );
 CapJitAddTypeSignature( "AsList", [ IsSkeletalFiniteSet ], rec( filter := IsList, element_type := rec( filter := IsInt ) ) );
 
+CapJitAddTypeSignature( "List", [ IsSkeletalFiniteSet, IsFunction ], rec( filter := IsList, element_type := rec( filter := IsInt ) ) );
```

Comment:
This is not sufficient:

1. This assumes that the function always maps to integers, which the implementation does not require/enforce.
2. You have to teach CompilerForCAP about the input type of the function so it can correctly type the function. This can be done with the following code (adapted from the existing `List` type signature):

```suggestion
CapJitAddTypeSignature( "List", [ IsSkeletalFiniteSet, IsFunction ], function ( args, func_stack )
    
    args := ShallowCopy( args );
    
    args.2 := CAP_JIT_INTERNAL_INFERRED_DATA_TYPES_OF_FUNCTION_BY_ARGUMENTS_TYPES( args.2, [ rec( filter := IsInt ) ], func_stack );
    
    if args.2 = fail then
        
        #Error( "could not determine output type" );
        return fail;
        
    fi;
    
    return rec( args := args, output_type := rec( filter := IsList, element_type := args.2.data_type.signature[2] ) );
    
end );
```

This code must live in `CompilerLogic` if we want to avoid a hard dependency in CompilerForCAP.

Reply:
Wonderful, thank you very much.